### PR TITLE
fuzz: Configure check for main function

### LINKED ISSUE #336: Error message "LLVMFuzzerTestOneInput is not defined" also when LLVMFuzzerTestOneInput is defined?
Thanks a lot for writing and maintaining a very nice fuzzer!

In Bitcoin Core we encourage our users to fuzz our code using various fuzzers including the excellent Honggfuzz.

These are the quick start instructions we provide at https://github.com/bitcoin/bitcoin/blob/master/doc/fuzzing.md:

```
$ git clone https://github.com/bitcoin/bitcoin
$ cd bitcoin/
$ ./autogen.sh
$ git clone https://github.com/google/honggfuzz
$ cd honggfuzz/
$ make
$ cd ..
$ CC=$(pwd)/honggfuzz/hfuzz_cc/hfuzz-clang CXX=$(pwd)/honggfuzz/hfuzz_cc/hfuzz-clang++ ./configure --enable-fuzz --with-sanitizers=address,undefined --enable-c++17
$ make
$ mkdir -p inputs/
$ honggfuzz/honggfuzz -i inputs/ -- src/test/fuzz/process_message
```

These instructions have been working fine up until quite recently when they [no longer appear to work when using a fresh `honggfuzz` compiled from `master`](https://github.com/bitcoin/bitcoin/pull/18739#issuecomment-629504388). (The only thing that has changed on our side is a recent switch to C++17 for the fuzzers, but I assume that is unrelated.)

The error message reported is `LLVMFuzzerTestOneInput():38 Define 'int LLVMFuzzerTestOneInput(uint8_t * buf, size_t len)' in your code to make it work`, but [we do define `LLVMFuzzerTestOneInput`](https://github.com/bitcoin/bitcoin/blob/54f812d9d29893c690ae06b84aaeab128186aa36/src/test/fuzz/fuzz.cpp#L34-L47) (and `libFuzzer` works as expected).

I assume our quick start instructions are incorrect in some way: are we missing something obvious here? :)

### PR DESCRIPTION
**Pull request by maflcko**
Instead of the PP jungle, use a proper configure check

Fixes https://github.com/google/honggfuzz/issues/336#issuecomment-702972138

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -1165,6 +1165,24 @@ if test "x$enable_fuzz" = "xyes"; then
...
+    [
+      AC_MSG_RESULT([yes])
+      CPPFLAGS="$CPPFLAGS -DPROVIDE_MAIN_FUNCTION"
+    ],
+    [],
+    [AC_LANG_PROGRAM([[
+    #include <cstdint>
+    #include <cstddef>
+    extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) { return 0; }
+    /* no main function ...
```
Comment (practicalswift) [inline configure.ac:1184]: `/*` not terminated? :)

**Comment by practicalswift** [comment]
Concept ACK

Thanks for fixing this!

File: configure.ac (hunk 1 above)
Comment (MarcoFalke) [reply configure.ac:1184]: Yes

Comment (practicalswift) [inline configure.ac:1184]: From a reviewer it looks odd that `/*` isn't terminated. Please add a comment if it is intentional :)

Comment (MarcoFalke) [reply configure.ac]: Thx. Made the comment longer

**Comment by practicalswift** [comment]
ACK fae7a1c18803675e70b9bf66575e1e0a6e01f6f6

**Pull request merged**